installers/server/release/server.sh

```diff
@@ -124,20 +162,17 @@ fi
 if [ "$USE_URANDOM" != "false" ] && [ -e "/dev/urandom" ]; then
     SERVER_STARTUP_ARGS+=("-Djava.security.egd=file:/dev/./urandom")
 fi
-CMD="$JAVA_HOME/bin/java ${SERVER_STARTUP_ARGS[@]} -jar $SERVER_DIR/go.jar"
 
-echo "Starting Go Server with command: $CMD" >> $STDOUT_LOG_FILE
-echo "Starting Go Server in directory: $GO_WORK_DIR" >> $STDOUT_LOG_FILE
-cd "$SERVER_WORK_DIR"
+RUN_CMD=("$(autoDetectJavaExecutable)" ${SERVER_STARTUP_ARGS[@]} "-jar" "$SERVER_DIR/go.jar")
 
-if [ "$JAVA_HOME" == "" ]; then
-    echo "Please set JAVA_HOME to proceed."
-    exit 1
-fi
+echo "[$(date)] Starting Go Server with command: ${RUN_CMD[@]}" >>"$STDOUT_LOG_FILE"
+echo "[$(date)] Starting Go Server in directory: $SERVER_WORK_DIR" >> $STDOUT_LOG_FILE
+cd "$SERVER_WORK_DIR"
 
 if [ "$DAEMON" == "Y" ]; then
-    eval exec nohup "$CMD" >> $STDOUT_LOG_FILE 2>&1 &
-    echo $! >$PID_FILE
+    "${RUN_CMD[*]}"  >> "$STDOUT_LOG_FILE" 2>&1 &
+    disown $!
+    echo $! >"$PID_FILE"
 else
-    eval exec "$CMD"
+    exec "${RUN_CMD[@]}"
```

Comment:
There's a bit of a difference between the eval and this, which you might have to think about. It has to do with how spaces are handled. See this:

```
$ cat args.sh
#!/bin/bash

echo "Args: $#"
for each in `seq $#`; do
  echo "Arg $each: $1"
  shift
done

$ ./args.sh a "bc de"
Args: 2
Arg 1: a
Arg 2: bc de

$ ARGS+=("arg1 arg2")
$ ARGS+=("arg3 \"arg4 with spaces\" arg5")
$ OTHER_ARGS="arg6 arg7 arg8"
$ ARGS+=("addotherargs $OTHER_ARGS")

$ # Old way in https://raw.githubusercontent.com/gocd/gocd/master/installers/server/release/server.sh
$ eval "./args.sh ${ARGS[@]}"
Args: 9
Arg 1: arg1
Arg 2: arg2
Arg 3: arg3
Arg 4: arg4 with spaces
Arg 5: arg5
Arg 6: addotherargs
Arg 7: arg6
Arg 8: arg7
Arg 9: arg8

$ # New way in this PR.
$ RUN_CMD=("./args.sh" ${ARGS[@]})
$ "${RUN_CMD[@]}"
Args: 11
Arg 1: arg1
Arg 2: arg2
Arg 3: arg3
Arg 4: "arg4
Arg 5: with
Arg 6: spaces"
Arg 7: arg5
Arg 8: addotherargs
Arg 9: arg6
Arg 10: arg7
Arg 11: arg8
```

This will probably cause issues in usage of `GO_SERVER_SYSTEM_PROPERTIES` (and `GO_AGENT_SYSTEM_PROPERTIES`).